commands/auth_test.go

```diff
@@ -14,228 +14,49 @@ limitations under the License.
+	assertCommandNames(t, cmd, "init")
 }
 
-func TestAuth_retrieveCredentials(t *testing.T) {
-	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		ac := doitserver.AuthCredentials{ID: "abc", CS: "def"}
-		err := json.NewEncoder(w).Encode(&ac)
-		if err != nil {
-			t.Fatal(err)
-		}
-	}))
-	defer ts.Close()
-
-	dsa := newDoitServerAuth()
-	dsa.url = ts.URL
-	ac, err := dsa.retrieveAuthCredentials()
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	expectedAC := &doitserver.AuthCredentials{ID: "abc", CS: "def"}
-	if got, want := ac, expectedAC; !reflect.DeepEqual(got, expectedAC) {
-		t.Fatalf("retrieveAuthCredentials() = %#v; got %#v", got, want)
-	}
-}
-
-func TestAuth_createAuthURL(t *testing.T) {
-	dsa := newDoitServerAuth()
-	dsa.url = "http://example.com"
-
-	ac := &doitserver.AuthCredentials{
-		ID: "id",
-		CS: "cs",
-	}
-	u, err := dsa.createAuthURL(ac)
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	newU, err := url.Parse(u)
-
-	if got, want := newU.Host, "example.com"; got != want {
-		t.Fatalf("createAuthURL() Host = %q, want = %q", got, want)
-	}
-
-	if got, want := newU.Path, "/auth/digitalocean"; got != want {
-		t.Fatalf("createAuthURL() Path = %q, want = %q", got, want)
-	}
-
-	q := newU.Query()
-
-	if got, want := q.Get("cs"), "cs"; got != want {
-		t.Fatalf("createAuthURL() cs param = %q, want = %q", got, want)
-	}
-
-	if got, want := q.Get("id"), "id"; got != want {
-		t.Fatalf("createAuthURL() id param = %q, want = %q", got, want)
-	}
-}
-
-func TestAuth_createAuthURL_with_keypairs(t *testing.T) {
-	dsa := newDoitServerAuth()
-	dsa.url = "http://example.com"
-	ac := &doitserver.AuthCredentials{
-		ID: "id",
-		CS: "cs",
-	}
-	u, err := dsa.createAuthURL(ac, keyPair{k: "foo", v: "bar"})
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	newU, err := url.Parse(u)
-
-	if got, want := newU.Host, "example.com"; got != want {
-		t.Fatalf("createAuthURL() Host = %q, want = %q", got, want)
-	}
-
-	if got, want := newU.Path, "/auth/digitalocean"; got != want {
-		t.Fatalf("createAuthURL() Path = %q, want = %q", got, want)
-	}
-
-	q := newU.Query()
-
-	if got, want := q.Get("cs"), "cs"; got != want {
-		t.Fatalf("createAuthURL() cs param = %q, want = %q", got, want)
-	}
-
-	if got, want := q.Get("id"), "id"; got != want {
-		t.Fatalf("createAuthURL() id param = %q, want = %q", got, want)
-	}
-
-	if got, want := q.Get("foo"), "bar"; got != want {
-		t.Fatalf("createAuthURL() foo param = %q, want = %q", got, want)
-	}
-}
-
-func TestAuth_initAuthCLI(t *testing.T) {
-	ogRUTF := retrieveUserTokenFunc
+func TestAuthInit(t *testing.T) {
+	rtf := retrieveUserTokenFunc
+	cfw := cfgFileWriter
 	defer func() {
-		retrieveUserTokenFunc = ogRUTF
+		retrieveUserTokenFunc = rtf
+		cfgFileWriter = cfw
 	}()
 
 	retrieveUserTokenFunc = func() (string, error) {
-		return "token", nil
+		return "valid-token", nil
 	}
 
-	dsa := newDoitServerAuth()
+	cfgFileWriter = func() (io.WriteCloser, error) { return &discardCloser{}, nil }
```

Comment:
actually, i was right. this is how you would do it. nopCloser wants an `io.Reader`